src/ios-scroll.js

```diff
@@ -0,0 +1,557 @@
+     * @private @const
+     */
+    this.boundAutoScroll_ = this.autoScroll_.bind(this);
+
+    /** @private @const {!Observable} */
+    this.scrollObservable_ = new Observable();
+
+    // We listen on the document level so that, even if the HTML or BODY has
+    // padding, margin, etc, the touch event will still cause scrolling.
+    doc.addEventListener('touchstart', this.touchstart_.bind(this));
+    doc.addEventListener('touchmove', this.touchmove_.bind(this));
+    doc.addEventListener('touchend', this.touchend_.bind(this));
+    // Native scrolling does some weird things. When the user native scrolls
+    // due to an iframe, the only event fired in this doc is a `scroll` on
+    // the body element.
+    body.addEventListener('scroll', this.bodyScroll_.bind(this));
+
+    // Promote the scroller to it's own GPU rendering layer.
+    this.scrollStyle_.willChange = 'transform';
+  }
+
+  /**
+   * Adds a fixed position element to the list we track. When the doc is
+   * scrolled, we will scroll the element in unison.
+   *
+   * @param {!Element} element
+   */
+  addFixedElement(element) {
+    const style = element.style;
+    this.fixedElementStyles_.push(style);
+    // Promote the element to it's own GPU rendering layer. This allows us to
+    // scroll the element in unison with the doc.
+    style.transform = `translate3d(0, 0, 0)`;
+  }
+
+  /**
+   * Removes the fixed position element from our list.
+   *
+   * @param {!Element} element
+   */
+  removeFixedElement(element) {
+    const style = element.style;
+    const index = this.fixedElementStyles_.indexOf(style);
+
+    if (index > -1) {
+      this.fixedElementStyles_.splice(index, 1);
+      style.transform = '';
+    }
+  }
+
+  /**
+   * Adds an iframe to the list we track. When the doc is scrolled, we will
+   * disable interaction with the element to prevent native scrolling from
+   * kicking in. When scrolling has stopped, we will re-enable.
+   *
+   * @param {!HTMLIFrameElement} iframe
+   */
+  addIframe(iframe) {
+    const style = iframe.style;
+    this.iframeStyles_.push(style);
+
+    // Offset is only present during scrolling, in which case we need to
+    // disable interacting with the iframe.
+    if (this.offset) {
+      style.pointerEvents = 'none';
+    }
+  }
+
+  /**
+   * Removes the iframe from our list.
+   *
+   * @param {!HTMLIFrameElement} iframe
+   */
+  removeIframe(iframe) {
+    const style = iframe.style;
+    const index = this.iframeStyles_.indexOf(style);
+
+    if (index > -1) {
+      this.iframeStyles_.splice(index, 1);
+      style.pointerEvents = '';
+    }
+  }
+
+  /**
+   * Scrolls the document to position ypos.
+   *
+   * @param {number} ypos
+   */
+  setScrollTop(ypos) {
+    this.scrollTo_(ypos);
+    this.scrollStopped_();
+  }
+
+  /**
+   * Calculates the current scrollTop.
+   */
+  getScrollTop() {
+    const delta = this.bodyScrolled_ ?
+        -this.scrollMoveEl_./*OK*/getBoundingClientRect().top  :
```

Comment:
Sigh, this is much more efficient than the current, which calls `#getBoundingClientRect` on every scroll event. Here, we only need this after the body have been natively scrolled (due to an iframe).